### PR TITLE
add additional ucsc tools

### PR DESCRIPTION
**Pull request by jianhong**
## PR checklist

- [ x ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ x ] If necessary, include test data in your PR.
- [ x ] Remove all TODO statements.
- [ x ] Emit the `<SOFTWARE>.version.txt` file.
- [ x ] Follow the naming conventions.
- [ x ] Follow the parameters requirements.
- [ x ] Follow the input/output options guidelines.
- [ x ] Add a resource `label`
- [ x ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ x ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ x ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: software/ucsc/bigwigaverageoverbed/main.nf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    input:
+    tuple val(meta), path(bed)
+    path bigwig
+
+    output:
+    tuple val(meta), path("*.tab")           , emit: tab
+    path "*.version.txt"                     , emit: version
+
+    script:
+    def software = "ucsc-bigWigAverageOverBed"
```
Comment (KevinMenden) [inline software/ucsc/bigwigaverageoverbed/main.nf:30]:
```suggestion
    def software = getSoftwareName(task.process)
```

File: software/ucsc/wigtobigwig/main.nf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    input:
+    path wig
+    path chromsizes
+
+    output:
+    path "*.bw"                   , emit: bw
+    path "*.version.txt"          , emit: version
+
+    script:
+    def software = 'ucsc-wigtobigwig'
```
Comment (KevinMenden) [inline software/ucsc/wigtobigwig/main.nf:30]:
```suggestion
    def software = getSoftwareName(task.process)
```

File: software/ucsc/wigtobigwig/meta.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      licence: ['varies; see http://genome.ucsc.edu/license']
+
+input:
+  - wig:
+      type: file
+      description: wig file
+      pattern: "*.{wig}"
+  - chromsizes:
+        type: file
+        description: chromosome sizes file
```
Comment (KevinMenden) [inline software/ucsc/wigtobigwig/meta.yml:24]:
```suggestion
      type: file
      description: chromosome sizes file
```

**Review by KevinMenden — APPROVED** [review]
Thank you, looks good to me!
If you could just use the common software version function, that would be great. I'll approve already anyway.
We try to keep it at one new module per PR, to make reviewing easer :)

**Comment by KevinMenden** [comment]
Also there's a slightly new verion of the `functions.nf` file available. If you could just update this as well, that would be great! (Just copy from some of the modules in nf-core/modules)

**Comment by jianhong** [comment]
> Also there's a slightly new verion of the `functions.nf` file available. If you could just update this as well, that would be great! (Just copy from some of the modules in nf-core/modules)

I want to double check about this. I tried to copy functions.nf file but did not see anything changed in code part except the comment style. Did I copy the correct file?

**Comment by KevinMenden** [comment]
Hi @jianhong, no that's alright, the only thing that changed is the comment style :-)  Maybe should have mentioned.

So functionally nothing changes, but the linter will complain that your `functions.nf` file is outdated and that's why it's good to update right away. Thanks!

**Pull request merged**